rollbar-web/src/main/java/com/rollbar/web/filter/RollbarFilter.java

```diff
@@ -0,0 +1,58 @@
+package com.rollbar.web.filter;
+
+import static com.rollbar.notifier.config.ConfigBuilder.withAccessToken;
+
+import com.rollbar.notifier.Rollbar;
+import com.rollbar.web.provider.PersonProvider;
+import com.rollbar.web.provider.RequestProvider;
+import java.io.IOException;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+
+public class RollbarFilter implements Filter {
+
+  private Rollbar rollbar;
```

Comment:
Does this need to be a `ThreadLocal`? Is a filter instantiated for each thread or request?

Reply:
AFAIK filters and servlets are one instance per application context, that's it per application deployed in the servlet container. 
At the same time, in this implementation we are using the library manage instance so the same instance will be used by every thread. The `rollbar` notifier should be thread-safe, thing that at this point I think it is if the `Providers` are. This should be worth mentioning in the `Provider` interface.